api/storage_service.cc

```diff
@@ -378,39 +377,16 @@ void unset_transport_controller(http_context& ctx, routes& r) {
     ss::is_native_transport_running.unset(r);
 }
 
-void set_rpc_controller(http_context& ctx, routes& r, thrift_controller& ctl) {
-    ss::stop_rpc_server.set(r, [&ctx, &ctl](std::unique_ptr<http::request> req) {
-        return smp::submit_to(0, [&] {
-            return with_scheduling_group(ctx.db.local().get_statement_scheduling_group(), [&ctl] {
-                return ctl.request_stop_server();
-            });
-        }).then([] {
-            return make_ready_future<json::json_return_type>(json_void());
-        });
-    });
-
-    ss::start_rpc_server.set(r, [&ctx, &ctl](std::unique_ptr<http::request> req) {
-        return smp::submit_to(0, [&] {
-            return with_scheduling_group(ctx.db.local().get_statement_scheduling_group(), [&ctl] {
-                return ctl.start_server();
-            });
-        }).then([] {
-            return make_ready_future<json::json_return_type>(json_void());
-        });
-    });
-
-    ss::is_rpc_server_running.set(r, [&ctl] (std::unique_ptr<http::request> req) {
-        return smp::submit_to(0, [&] {
-            return !ctl.listen_addresses().empty();
-        }).then([] (bool running) {
-            return make_ready_future<json::json_return_type>(running);
+// NOTE: preserved only for backward compatibility
+void set_rpc_controller(http_context& ctx, routes& r) {
+    ss::is_rpc_server_running.set(r, [] (std::unique_ptr<http::request> req) {
```

Comment:
One of the most annoying things about Thrift was how it referred to itself as "RPC", confusing with our other "RPC" (the one we actually care about).
How hard will it be NOT to preserve this Thrift "RPC" stuff and this confusing archaic name?

Reply:
it's not that hard. the point is to be backward compatible with the java-based nodetool, it is still calling this API. if we keep it around, it should work. i don't want to have a tool installed and claim it broken.

Reply:
Maybe we should have at least renamed this function to something less confusing (since we usually use the word "RPC" for something else...), or added a clarifying comment that this is about deprecated Thrift or something. But never mind, I don't insist.